src/card.tsx

```diff
@@ -276,34 +369,62 @@ export const DataCard = React.forwardRef<HTMLDivElement, DataCardProps>(
         },
         ref
     ) => {
+        const finalActions = useTopActions(actions, onClose);
+        const hasActions = finalActions.length > 0;
+
         return (
```

Comment:
About the right space to avoid collision between card content and top actions:
- I wouldn't add any space when there is an icon (the collision in that case is almost imposible)
- `extra` should always be rendered ful width, like the buttons.

I know this will make this layout logic a bit more complex, but I think it's worth it